tests/tick.test.mjs

```diff
@@ -659,4 +664,121 @@ describe('Tick', async () => {
+      memory: genesisEpochTick.Memory,
+    });
+
+    // Buy records in this epoch
+    let buyRecordMemory = transferMemory;
+    for (let i = 0; i < 10; i++) {
+      const { Memory } = await handle(
+        {
+          Tags: [
+            { name: 'Action', value: 'Buy-Record' },
+            { name: 'Name', value: 'test-name-' + i },
+            { name: 'Purchase-Type', value: 'permabuy' },
+          ],
+        },
+        buyRecordMemory,
+      );
+      buyRecordMemory = Memory;
+    }
+
+    // Tick to the half way through the first epoch
+    const firstEpochMidTimestamp = genesisEpochStart + epochDurationMs / 2;
+    const firstEpochMidTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: firstEpochMidTimestamp,
+      },
+      buyRecordMemory,
+    );
+    const feeDuringFirstEpoch = await getBaseRegistrationFee(
+      firstEpochMidTick.Memory,
+    );
+
+    assert.equal(feeDuringFirstEpoch, 600_000_000);
+    const firstEpochDemandFactorResult = await getDemandFactor(
+      firstEpochMidTick.Memory,
+    );
+    assert.equal(firstEpochDemandFactorResult, 1);
+
+    // Tick to the end of the first epoch
+    const firstEpochEndTimestamp =
+      genesisEpochStart + epochDurationMs + distributionDelayMs + 1;
+    const firstEpochEndTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: firstEpochEndTimestamp,
+      },
+      buyRecordMemory,
+    );
+    const feeAfterFirstEpochEnd = await getBaseRegistrationFee(
+      firstEpochEndTick.Memory,
+    );
+
+    assert.equal(feeAfterFirstEpochEnd, 630_000_000);
+
+    const firstEpochEndDemandFactorResult = await getDemandFactor(
+      firstEpochEndTick.Memory,
+    );
+    assert.equal(firstEpochEndDemandFactorResult, 1.0500000000000000444);
+  });
+
+  it('should reset to baseRegistrationFee when demandFactor is 0.5 for consecutive epochs', async () => {
+    const zeroEpochTick = await handle(
+      {
+        Tags: [{ name: 'Action', value: 'Tick' }],
+        Timestamp: genesisEpochStart,
+      },
+      startMemory,
+    );
+
+    const baseFeeAtZeroEpoch = await getBaseRegistrationFee(
+      zeroEpochTick.Memory,
+    );
+    assert.equal(baseFeeAtZeroEpoch, 600_000_000);
+
+    let tickMemory = zeroEpochTick.Memory;
+    // Tick to the forth epoch,
+    for (let i = 0; i < 4; i++) {
+      const epochTimestamp = genesisEpochStart + (epochDurationMs + 1) * i;
+      const { Memory } = await handle(
+        {
+          Tags: [
+            { name: 'Action', value: 'Tick' },
+            { name: 'Timestamp', value: epochTimestamp.toString() },
+          ],
+          Timestamp: epochTimestamp,
+        },
+        tickMemory,
+      );
+
+      tickMemory = Memory;
+    }
+
+    const baseFeeAfterConsecutiveTicksWithNoPurchases =
+      await getBaseRegistrationFee(tickMemory);
+    assert.equal(baseFeeAfterConsecutiveTicksWithNoPurchases, 573_402_975);
```

Comment:
add check that the demand factor was reset to 1 after adjusting fees

Reply:
demand factor after 4 ticks is `0.95567162500000002456`

Reply:
for 20 ticks, it is `0.750392318118884738`

Reply:
we need 49 ticks from genesis to get back to 1 demand factor. the fee remains at 300_000_000 however

<img width="929" alt="image" src="https://github.com/user-attachments/assets/d0efcc06-32ef-4372-83dd-63918a1b6ae9">

91d1ba0